src/core/permissions/PermissionsSagas.js

```diff
@@ -215,9 +219,46 @@ function* getStudyAuthorizationsWatcher() :Generator<*, *, *> {
   yield takeEvery(GET_STUDY_AUTHORIZATIONS, getStudyAuthorizations);
 }
 
+function* getDeletePermissionWorker(action :SequenceAction) :Saga<*> {
+  try {
+    yield put(getDeletePermission.request(action.id));
+
+    // to delete study/participant data, current user should have OWNER on all the entity set ids
+    const coreEntitySetIds = yield select(selectEntitySetsByModule(AppModules.CHRONICLE_CORE));
+    const dataCollectionEntitySetids = yield select(selectEntitySetsByModule(AppModules.DATA_COLLECTION));
+    const surveyEntitySetIds :Map = yield select(selectEntitySetsByModule(AppModules.QUESTIONNAIRES));
+
+    const allEntityKeyIds = coreEntitySetIds
+      .merge(dataCollectionEntitySetids)
+      .merge(surveyEntitySetIds)
+      .valueSeq()
+      .toJS();
+
+    const acls = allEntityKeyIds.map((ekid) => [ekid]);
+
+    const response = yield call(getAclsWorker, getAcls(acls));
```

Comment:
No change required. Since we don't need the response for this request, perhaps we should add a new endpoint to `PermissionsApi` that only does the `ensurePermission` check. That being said, an endpoint like this might be a problem from a security perspective 🤔 .